winch/codegen/src/isa/aarch64/masm.rs

```diff
@@ -159,21 +159,27 @@ impl Masm for MacroAssembler {
     }
 
     fn load(&mut self, src: Address, dst: Reg, size: OperandSize) {
-        self.asm.ldr(src, dst, size);
+        self.asm.ldr(src, dst, size, false);
     }
 
-    fn load_ptr(&mut self, _src: Self::Address, _dst: Reg) {
-        todo!()
+    fn load_ptr(&mut self, src: Self::Address, dst: Reg) {
+        self.load(src, dst, self.ptr_size);
     }
 
     fn wasm_load(
         &mut self,
-        _src: Self::Address,
-        _dst: Reg,
-        _size: OperandSize,
-        _kind: Option<ExtendKind>,
+        src: Self::Address,
+        dst: Reg,
+        size: OperandSize,
+        kind: Option<ExtendKind>,
     ) {
-        todo!()
+        self.asm.ldr(
+            src,
+            dst,
+            size,
+            kind.is_some(), // kind is some if the value is signed
```

Comment:
Could `ldr` instead of accepting a boolean accept `Option<ExtKind>`? Or instead -- and maybe this is what I'd prefer -- we could introduce two methods at the assembler level:

- `sload`
- `uload`

And we could dispatch from `wasm_load` to the corresponding method depending on the value of `kind`, this would make more explicit depending on the extension kind. Additionally this would make it less error prone than passing a boolean.